specification/security/resource-manager/Microsoft.Security/stable/2024-08-01/customRecommedations.json

```diff
@@ -0,0 +1,473 @@
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/CustomRecommendation"
+            }
+          },
+          "200": {
+            "description": "OK - Updated",
+            "schema": {
+              "$ref": "#/definitions/CustomRecommendation"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "delete": {
+        "x-ms-examples": {
+          "Delete a custom recommendation over subscription scope": {
+            "$ref": "./examples/CustomRecommendations/DeleteBySubscriptionCustomRecommendation_example.json"
+          },
+          "Delete a custom recommendation over security connector scope": {
+            "$ref": "./examples/CustomRecommendations/DeleteBySecurityConnectorCustomRecommendation_example.json"
+          },
+          "Delete a custom recommendation over management group scope": {
+            "$ref": "./examples/CustomRecommendations/DeleteByManagementGroupCustomRecommendation_example.json"
+          }
+        },
+        "tags": [
+          "CustomRecommendations"
+        ],
+        "description": "Delete a custom recommendation over a given scope",
+        "operationId": "CustomRecommendations_Delete",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/Scope"
+          },
+          {
+            "$ref": "#/parameters/customRecommendationName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - custom recommendation was deleted"
+          },
+          "204": {
+            "description": "No Content - custom recommendation does not exist"
+          },
+          "default": {
+            "description": "Error response that describes why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CustomRecommendationsList": {
+      "type": "object",
+      "description": "A list of Custom Recommendations",
+      "properties": {
+        "value": {
+          "description": "Collection of Custom Recommendations",
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CustomRecommendation"
+          }
+        },
+        "nextLink": {
+          "description": "The link used to get the next page of operations.",
+          "type": "string"
+        }
+      }
+    },
+    "CustomRecommendation": {
```

Comment:
>CustomRecommendation [](http://example.com/codeflow?start=5&length=20)

The request and response bodies look very similar (identical?). Did you consider using the same definition for both? That is the usual approach.

Reply:
They are not identical, the response includes additional property (assessmentKey)

Reply:
So what happens if a client does GET, saves the body, modifies some properties, and then does PUT of the [modified] body that was returned by GET? Will it work, or will they get an error if they don't remove the assessmentKey property? The above sequence is a canonical ARM pattern and should be supported. The assessmentKey property would be optional in the request body by default.

Reply:
Please address this too